provider/resource_keycloak_openid_client_authorization_aggregate_policy.go

```diff
@@ -0,0 +1,185 @@
+		Name:             data.Get("name").(string),
+		Type:             "aggregate",
+		Policies:         policies,
+		Resources:        resources,
+		Scopes:           scopes,
+		Description:      data.Get("description").(string),
+	}
+	return &resource
+}
+
+func setOpenidClientAuthorizationAggregatePolicyResourceData(data *schema.ResourceData, policy *keycloak.OpenidClientAuthorizationAggregatePolicy) {
+	data.SetId(policy.Id)
+
+	data.Set("resource_server_id", policy.ResourceServerId)
+	data.Set("realm_id", policy.RealmId)
+	data.Set("name", policy.Name)
+	data.Set("decision_strategy", policy.DecisionStrategy)
+	data.Set("owner", policy.Owner)
+	data.Set("logic", policy.Logic)
+	data.Set("policies", policy.Policies)
+	data.Set("resources", policy.Resources)
+	data.Set("scopes", policy.Scopes)
+	data.Set("description", policy.Description)
+}
+
+func resourceKeycloakOpenidClientAuthorizationAggregatePolicyCreate(data *schema.ResourceData, meta interface{}) error {
+	keycloakClient := meta.(*keycloak.KeycloakClient)
+
+	resource := getOpenidClientAuthorizationAggregatePolicyResourceFromData(data)
+	err := keycloakClient.NewOpenidClientAuthorizationAggregatePolicy(resource)
+	if err != nil {
+		return err
+	}
+
+	setOpenidClientAuthorizationAggregatePolicyResourceData(data, resource)
+
+	return resourceKeycloakOpenidClientAuthorizationAggregatePolicyRead(data, meta)
+}
+
+func resourceKeycloakOpenidClientAuthorizationAggregatePolicyRead(data *schema.ResourceData, meta interface{}) error {
+	keycloakClient := meta.(*keycloak.KeycloakClient)
+
+	realmId := data.Get("realm_id").(string)
+	resourceServerId := data.Get("resource_server_id").(string)
+	id := data.Id()
+
+	resource, err := keycloakClient.GetOpenidClientAuthorizationAggregatePolicy(realmId, resourceServerId, id)
+	if err != nil {
+		return handleNotFoundError(err, data)
+	}
+
+	setOpenidClientAuthorizationAggregatePolicyResourceData(data, resource)
+
+	return nil
+}
+
+func resourceKeycloakOpenidClientAuthorizationAggregatePolicyUpdate(data *schema.ResourceData, meta interface{}) error {
+	keycloakClient := meta.(*keycloak.KeycloakClient)
+	resource := getOpenidClientAuthorizationAggregatePolicyResourceFromData(data)
+
+	err := keycloakClient.UpdateOpenidClientAuthorizationAggregatePolicy(resource)
+	if err != nil {
+		return err
+	}
+
+	setOpenidClientAuthorizationAggregatePolicyResourceData(data, resource)
+
+	return nil
+}
+
+func resourceKeycloakOpenidClientAuthorizationAggregatePolicyDelete(data *schema.ResourceData, meta interface{}) error {
+	keycloakClient := meta.(*keycloak.KeycloakClient)
+
+	realmId := data.Get("realm_id").(string)
+	resourceServerId := data.Get("resource_server_id").(string)
+	id := data.Id()
+
+	return keycloakClient.DeleteOpenidClientAuthorizationAggregatePolicy(realmId, resourceServerId, id)
+}
+
+func resourceKeycloakOpenidClientAuthorizationAggregatePolicyImport(d *schema.ResourceData, _ interface{}) ([]*schema.ResourceData, error) {
```

Comment:
It looks like this import function is basically the same as all of the other import functions for the policy resources - could you reuse the same function for all of them if possible?

Take a look at how I did this with protocol mappers: https://github.com/mrparkers/terraform-provider-keycloak/blob/master/provider/generic_protocol_mapper_helpers.go